public/components/datasources/components/manage/accelerations/acceleration_details_flyout.tsx

```diff
@@ -24,20 +24,90 @@ import {
-} from '../accelerations/helpers/utils';
+} from './utils/acceleration_utils';
+import { coreRefs } from '../../../../../framework/core_refs';
+import { OpenSearchDashboardsResponse } from '../../../../../../../../src/core/server/http/router';
 
 export interface AccelerationDetailsFlyoutProps {
   acceleration: any;
 }
 
-export const AccelerationDetailsFlyout = (props: AccelerationDetailsFlyoutProps) => {
-  const { acceleration } = props;
+const getMappings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  console.log('getMappings index: ', index);
+  return coreRefs.dslService?.fetchFields(index);
+};
+
+const getSettings = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchSettings(index);
+};
+
+const getIndexInfo = (index: string): Promise<OpenSearchDashboardsResponse> | undefined => {
+  return coreRefs.dslService?.fetchIndices(index);
+};
+
+const handlePromise = (
+  promise: Promise<OpenSearchDashboardsResponse> | undefined,
+  action: string
+) => {
+  return promise!
+    .then((data) => ({ status: 'fulfilled', action, data }))
+    .catch((error) => ({ status: 'rejected', action, error }));
+};
+
+export const AccelerationDetailsFlyout = ({
+  acceleration: selectedAcc,
+}: AccelerationDetailsFlyoutProps) => {
+  const { index, dataSourceName, acceleration } = selectedAcc;
+  const { flintIndexName } = acceleration;
   const [selectedTab, setSelectedTab] = useState('details');
   const tabsMap: { [key: string]: any } = {
     details: AccelerationDetailsTab,
     schema: AccelerationSchemaTab,
     sql_definition: AccelerationSqlTab,
   };
+  const [settings, setSettings] = useState<object>();
+  const [mappings, setMappings] = useState();
+  const [indexInfo, setIndexInfo] = useState();
+
+  const updateMapping = (result) => {
+    console.log('updateMapping: ');
+    console.log(result);
+    setMappings(result);
+  };
+
+  const updateSetting = (result, slectedIndex: string) => {
+    console.log('updateSetting: ');
+    console.log(result);
+    setSettings(result.data[slectedIndex]);
+  };
+
+  const updateIndexInfo = (result) => {
+    console.log('updateIndexInfo: ');
+    console.log(result);
+    setIndexInfo(result);
+  };
+
+  const getAccDetail = (selectedIndex: string) => {
+    Promise.all([
+      handlePromise(getMappings(selectedIndex), 'getMappings'),
+      handlePromise(getSettings(selectedIndex), 'getSettings'),
+      handlePromise(getIndexInfo(selectedIndex), 'getIndexInfo'),
+    ])
+      .then((results) => {
+        updateMapping(results[0]);
+        updateSetting(results[1], selectedIndex);
+        updateIndexInfo(results[2]);
+      })
+      .catch((errors: Error[]) => {
+        errors.forEach((error, errorIndex) => {
+          console.error(`Error in async call ${errorIndex + 1}:`, error);
+        });
+      });
+  };
+
+  useEffect(() => {
+    getAccDetail(flintIndexName);
+  }, [flintIndexName]);
```

Comment:
missing dependency 'getAccDetail', and question: do we need to handle cases where flintIndexName can be '' or undefined?

Reply:
I think we don't need to handle that case, since the field `flintIndexName` field should be the actual 'index name' in OS, so that it should be always present as long as the index existed.